File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/order/dtos/OrderDetailDTO.java
package vn.com.ltdt.Coffee_Shop.order.dtos;

import vn.com.ltdt.Coffee_Shop.product.dtos.ProductDTO;
import vn.com.ltdt.Coffee_Shop.product.dtos.ProductDetailDTO;

public record OrderDetailDTO(
        ProductDetailDTO productDetail,
        ProductDTO product,
        int quantity,
        double price
) {
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/remote/OrderDetailDTO.kt
package com.ltdt.coffeeshop_android_native.data.remote

import com.ltdt.coffeeshop_android_native.data.domains.OrderDetail

data class OrderDetailDTO(
    val productDetail: ProductDetailDTO,
    val product: ProductDTO,
    val quantity: Int,
    val price: Double
)

fun OrderDetailDTO.toEntity() = OrderDetail(
    productDetail = productDetail.toEntity(),
    product = product.toEntity(),
    quantity = quantity,
    price = price
)

